Rubberduck.CodeAnalysis/QuickFixes/RefactoringQuickFixBase.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using Rubberduck.Inspections.Abstract;
+using Rubberduck.Parsing.Inspections.Abstract;
+using Rubberduck.Parsing.Rewriter;
+using Rubberduck.Refactorings;
+using Rubberduck.Refactorings.Exceptions;
+
+namespace Rubberduck.Inspections.QuickFixes
+{
+    public abstract class RefactoringQuickFixBase : QuickFixBase
+    {
+        protected readonly IRefactoring Refactoring;
+
+        protected RefactoringQuickFixBase(IRefactoring refactoring, params Type[] inspections)
+            : base(inspections)
+        {
+            Refactoring = refactoring;
+        }
+
+        //The rewriteSession is optional since it is not used in refactoring quickfixes.
+        public override void Fix(IInspectionResult result, IRewriteSession rewriteSession = null)
+        {
+            try
+            {
+                Refactor(result);
+            }
+            catch (RefactoringAbortedException)
+            { }
+            catch (RefactoringException exception)
+            {
+                //This is an error: the inspection returned an invalid result. 
+                Logger.Error(exception);
+            }
+        }
+
+        protected abstract void Refactor(IInspectionResult result);
+
+        public override bool CanFixInProcedure => false;
+        public override bool CanFixInModule => false;
+        public override bool CanFixInProject => false;
```

Comment:
The point is that refactoring quick fixes use an `IRewriteSession` freshly requested by the corresponding refactoring instead of the one passed to the quick fix. The refactoring calls `Rewrite` on this `IRewriteSession` as part of applying the refactoring. This in turn causes a reparse, which Invalides the inspection results.

Accordingly, refactoring quickfixes can only be run separately.